### ISSUE TITLE
WebFluxLinkBuilder disregards path set in exchange

### ISSUE THREAD

**Issue by ctytgat** [spring-projects/spring-hateoas#1150]
Version 1.0.1

WebFluxLinkBuilder doesn't seem to take the path from the current exchange into account. 

See this method at the end of the class:
```
private static Function<String, UriComponentsBuilder> getBuilderCreator(UriComponentsBuilder exchange) {
   return path -> exchange.replacePath(path == null ? "/" : path);
}
``` 

The full request path is replaced by just the controller mapping, so some information is potentially lost. For example these 2 use cases don't generate correct links:

- x-forwarded-prefix header (processed by ForwardedHeaderTransformer)
- setting a context path https://stackoverflow.com/questions/49196368/context-path-with-webflux

**Comment by odrotbohm** [comment]
I guess we're gonna need some input from the WebFlux team here.

**Comment by rstoyanchev** [comment]
The contextPath is available from `ServerHttpRequest` via `getPath().contextPath()`, and `ForwardedHeaderTransformer` sets the `contextPath` too if it finds a prefix.

**Comment by odrotbohm** [comment]
That's now fixed and backported to 1.0.x.
